xs2a-adapter-aspsp-registry/src/test/java/de/adorsys/xs2a/adapter/registry/LuceneAspspRepositoryTest.java

```diff
@@ -196,6 +197,70 @@ public void findLike() {
         assertThat(found).hasSize(2);
     }
 
+    @Test
+    public void findLikeShouldFollowTheStrictOrder() {
+        List<Aspsp> expected = Arrays.asList(
+            buildAspsp("TESTBICA", "111111", "SomeBank"),
+            buildAspsp("TESTBICA", "111111"),
+            buildAspsp("TESTBICA", "111111"),
+            buildAspsp("TESTBICA", "111111"),
+            buildAspsp("TESTBICA", "111111"),
+            buildAspsp("TESTBICA", "111111"),
+            buildAspsp("TESTBICA", null),
+            buildAspsp("TESTBICA", null),
+            buildAspsp(null, "111111"),
+            buildAspsp(null, "111111")
+        );
+
+        luceneAspspRepository.save(buildAspsp("TESTBICA", "111111"));
+        luceneAspspRepository.save(buildAspsp("TESTBICA", "111111"));
+        luceneAspspRepository.save(buildAspsp("TESTBICA", "111111"));
+        luceneAspspRepository.save(buildAspsp("TESTBICA", "111111"));
+        luceneAspspRepository.save(buildAspsp("TESTBICA", "111111"));
+        luceneAspspRepository.save(buildAspsp("TESTBICB", "222222"));
+        luceneAspspRepository.save(buildAspsp("TESTBICC", "333333"));
+        luceneAspspRepository.save(buildAspsp("TESTBICD", "444444"));
+        luceneAspspRepository.save(buildAspsp("TESTBICE", "555555"));
+        luceneAspspRepository.save(buildAspsp(null, "111111"));
+        luceneAspspRepository.save(buildAspsp(null, "111111"));
+        luceneAspspRepository.save(buildAspsp(null, "111111"));
+        luceneAspspRepository.save(buildAspsp(null, "111111"));
+        luceneAspspRepository.save(buildAspsp("TESTBICA", null));
+        luceneAspspRepository.save(buildAspsp("TESTBICA", null));
+        luceneAspspRepository.save(buildAspsp("TESTBICA", "111111", "SomeBank"));
+
+        List<Aspsp> actual = luceneAspspRepository.findLike(buildAspsp("TESTBICA", "111111", "SomeBank"));
+
+        assertThat(actual).hasSize(10);
+        assertThatOrderIsTheSame(actual, expected);
```

Comment:
`isEqualTo` should suffice

Reply:
Unfortunately not, as the Aspsp-s from the Lucene contain also `id` and `paginationId` filled, so `isEqualTo()` wouldn't work

Reply:
I see. `paginationId` and `id` should not be considered in equals/hashcode since they are not manually assigned. The same principle as for JPA entities.
The fact that we have autogeneration of id's is a problem in and of itself.

Reply:
fixed, thanks